README.md

```diff
@@ -1 +1,318 @@
-# sql-joins
+# SQL Joins
+
+## Objectives
+
+1. Define SQL joins and understand when and why to use them.
+2. Define the different types of SQL joins: inner, outer, left outer, right outer
+3. Practice writing join statements. 
+
+## What Is a JOIN?
+
+A SQL JOIN clause is a way to combine rows from two or more tables, based on a common column between them. The great thing about relational databases is that they are just that––*relational*. Relational databases allow us not only to store data that is interconnected, but to retrieve that data in ways that reflect that interconnectivity. 
+
+Let's say, for example, we have two tables, a Cats table and an Owners table. Cats and owners are associated by a foreign key of `owner_id` in the Cats table. How would we craft a query that would grab us all of the cats with a particular owner, and even include information about that owner in the data returned to us by that query? We know how to write a `SELECT` statement that gets us all of the cats with a particular `owner_id`. For example:
```

Comment:
you jump right into it too quickly. With `foreign_key` and all that. They have no idea what `foreign_key` is yet! I would start with something about at a pretty abstract level what relations are trying to solve. So maybe you are creating a pet database (ALSO! I love when we re-use domains from previous labs, decreases the cognitive load a ton. Also tells a bit more of the story :) )and you  then have owners and pets. You have a problem. There could be a crazy cat lady who has more then one cat! Given what we know about databases, how would we create that? Doesn't seem possible....how would we do this in just pure ruby? We'd want some sort of like array attribute on the `Owner` class that has all the pets. Hmmm but what would be in that array? It would be an array of `Pet` objects. In general you can think if you would make a new class for something, you should make a new table. So now we have two tables in SQL...but we need to *relate* them. Each row can only contain one element per column. In our owners table, we can't just list all of the pets, that's more hten one item. But! How many owners does a cat have? one. We can do one. Since all rows have that `id` column and that is the unique identifier for each row. We can create an `owner_id` field in the pet table to figure out which owner, owns that pet. In DB speak, we call this a `foreign_key` because it is "foreign" to the pet table.